test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -129,6 +148,50 @@ prop_applyBlockBasic s =
             (totalBalance wallet === balance utxo')
 
 
+prop_applyBlockTxHistoryIncoming :: WalletState -> Property
+prop_applyBlockTxHistoryIncoming s = conjoin (map prop wallets)
+  where
+    wallets = scanl (flip applyBlock) (initWallet s) blockchain
+    -- each transaction must have at least one output belonging to us
+    prop wallet = conjoin [addrs `overlaps` ourAddresses s | addrs <- txOuts]
+      where
+        inTxs = filter isIncoming $ Set.toList $ getTxHistory wallet
+        isIncoming (_, m) = direction m == Incoming
+        txOuts = map (Set.fromList . map address . outputs . fst) inTxs
+        overlaps a b = not (Set.disjoint a b)
```

Comment:
It'll be nice to add a coverage checks that makes sure that `addrs` and `ourAddresses` aren't just empty all the time.